src/pubsub/ua_pubsub_networkmessage.h

```diff
@@ -145,22 +140,17 @@ typedef struct {
     UA_Boolean publisherIdEnabled;
     UA_Boolean groupHeaderEnabled;
     UA_Boolean payloadHeaderEnabled;
-    UA_PublisherIdDatatype publisherIdType;
     UA_Boolean dataSetClassIdEnabled;
     UA_Boolean securityEnabled;
     UA_Boolean timestampEnabled;
     UA_Boolean picosecondsEnabled;
     UA_Boolean chunkMessage;
     UA_Boolean promotedFieldsEnabled;
     UA_NetworkMessageType networkMessageType;
-    union {
-        UA_Byte publisherIdByte;
-        UA_UInt16 publisherIdUInt16;
-        UA_UInt32 publisherIdUInt32;
-        UA_UInt64 publisherIdUInt64;
-        UA_Guid publisherIdGuid;
-        UA_String publisherIdString;
-    } publisherId;
+    UA_PublisherIdType publisherIdType;
+    UA_PublisherId publisherId; /* publisherId is a shallow copy of the PublisherId from connection configuration
```

Comment:
Contrary to the comment, the string id is cleaned in UA_NetworkMessage_clear:
[src/pubsub/ua_pubsub_networkmessage.c](https://github.com/open62541/open62541/pull/5223/files#diff-472e00c38461c6689d68bb19d3b179db332aca56729cebc9af2339f12f517afd)

If you want this to be a shallow copy I suggest you manually call UA_String_init on the publisher id outside of UA_NetworkMessage_clear. That way the information that the id is a shallow copy is handled in one location and is not distributed throughout the stack.

Pseudo-Code:
```c
message->id = config->id; // shallow copy. reset before calling UA_NetworkMessage_clear
send_message(message);
if(is_string)
   UA_String_init(&message->id.string);
UA_NetworkMessage_clear(message);
```

Then this comment can be removed.

Reply:
The way I understand it is that in fact publisherId is a deep copy (at least for strings) because of UA_String_decodeBinary in ua_pubsub_networkmessage.c line 604. So the comment is wrong and should be removed. The cleaning of the string in UA_NetworkMessage_clear is necessary.

Reply:
Removed comment